### ISSUE TITLE
Stuck in the APNS sandbox

### ISSUE THREAD

**Issue by paul-doherty** [aws-amplify/amplify-js#2577]
** Which Category is your question related to? **
Push Notifications

** What AWS Services are you utilizing? **
Pinpoint

** Provide additional details e.g. code snippets **
I have setup my React Native project according to the guide and it partly works for iPhones. This **only** works if I select the APNS sandbox. How am I supposed to toggle sandbox usage or not and use production?

I have configured Pinpoint to use a key **not** a certificate, so it should be able to hit prod and the sandbox. 

Thanks

**Comment by RobertNg1125** [comment]
Just want to add one more thing: 

- If you are testing by using Test messaging in Pinpoint dashboard (build with **Release**, get device token, open Test messaging, paste in device token ... ), that would work right away
- If you are testing by sending Pinpoint campaign, make sure to verify that your endpoint shown up correctly in the target segment. There is chance that you don't register correctly to the segment -> no notification

I've been wrestling with Pinpoint these days and learnt these on the way
Hope this help,

**Comment by paul-doherty** [comment]
@RobertNg1125 A sincere thanks, you saved me from myself.

When you were testing in dev mode did Test Messaging work for you when targeting by EndpointId? It errors for me but gives me the correct token in the error response. I can take that token and change the destination type to token, service to APNS Sandbox and it works. It seems like Pinpoint isn't selecting APNS Sandbox for the service though it looks like it should detect it automatically or at least allow me to set and attribute on the endpoint to specify which is desired.

**Comment by RobertNg1125** [comment]
@paul-doherty I have never send Test messaging using EndpointID ... 

I'm not sure if it's your case or not, but in my case, when I tested my app, I kept installing/removing the test app multiple times, and I use device token for the "Address" attribute of the endpoint. It created different Cognito Identity ID each time, even though the device token is the same. 

By exporting the list of endpoints, I noticed more than one endpoint have same device token, some with APNS and some with APNS Sandbox. 

My solution was to wipe out all the Cognito Identity ID related to my device token, then all behave normally. 

My advice would be: export list of your endpoints and you'll have better insight on your endpoints, which help you to channel the push notification better. 

Good luck : )
